lib/mail/notify/mailer.rb

```diff
@@ -2,19 +2,94 @@
 
 module Mail
   module Notify
+    ##
+    # The Mail Notify base Mailer class, overridden in Rails applications to provide the additional
+    # Notify behaviour along with the application behaviour.
+
     class Mailer < ActionMailer::Base
-      def view_mail(template_id, headers)
-        raise ArgumentError, "You must specify a template ID" if template_id.blank?
+      ##
+      # Set a default from address, will only be used in previews if a from address is not supplied
+      # by subclasses
+
+      default from: "preview@notifications.service.gov.uk"
+
+      ##
+      # Send an email where the content is managed in the Notify template.
+      #
+      # The required arguments are:
+      #
+      # - template_id
+      # - to address
+      #
+      # Can include personalisation.
+      #
+      # Add any additional headers in the options hash.
+      #
+      # A default subject is supplied as ActionMailer requires one, however it will never be used as
+      # the subject is assumed to be managed in the Notify template.
+
+      def template_mail(template_id, options)
+        raise ArgumentError, "You must specify a Notify template ID" if template_id.blank?
+        raise ArgumentError, "You must specify a to address" if options[:to].nil? || options[:to].blank?
+
+        message.template_id = template_id
+        message.reply_to_id = options[:reply_to_id]
+        message.reference = options[:reference]
+
+        message.personalisation = options[:personalisation] || {}
+
+        headers = options.except([:personalisation, :reply_to_id, :reference])
 
-        mail(headers.merge(template_id: template_id))
+        headers[:subject] = "Subject managed in Notify" unless options[:subject]
+
+        mail(headers) do |format|
+          format.text { nil }
+          format.html { nil }
+        end
```

Comment:
I wonder if some comments here about why we're setting blank headers would be nice here